nkms/crypto/keypairs.py

```diff
@@ -125,6 +127,23 @@ def _vrs_msgpack_load(self, msgpack_vrs):
         s = int.from_bytes(sig[2], byteorder='big')
         return (v, r, s)
 
+    def digest(self, data: Iterable) -> bytes:
+        """
+        Accepts an iterable containing bytes and digests it.
+
+        :param Iterable data: Data to digest
+
+        :rtype: bytes
+        :return: bytestring of digested data
+        """
+        if type(data) == bytes:
+            return sha3.keccak_256(data).digest()
+
+        hash = sha3.keccak_256()
+        for item in data:
```

Comment:
This line will do some things that are perhaps unexpected:

* For a non-iterable, it will raise "TypeError: '{whatever}'' object is not iterable"
* For a string - instead of casting to bytes, it will iterate through the string and update the hash.  Is that what we want?  Maybe it is.

Reply:
A `String` can't be used in the hash digest, it needs to be of type `bytes`. Most APIs hand this off to the user to handle their own encoding.

If it raises a `TypeError`, I would completely consider that expected behavior.

Reply:
Well, as I said, it won't with this implementation.  It will iterate through the string.

Reply:
Oh, right, and then I guess it will raise TypeError in the next line.

Reply:
So I guess that's fine.  But I think that the "TypeError: '{whatever}'' object is not iterable" is a little strange - better to have "TypeError: message must be bytes or an collection of bytes."

Reply:
It has almost exactly that error message:

```
>>> sha3.keccak_256('hash a string').digest()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unicode-objects must be encoded before hashing
```

Reply:
Sure - string is fine.  But what if someone passes `int`?

Reply:
Similar `TypeError`, but just informs the user that it requires an object that supports the buffer API.